### PR TITLE
Update GraphQL documentation

### PR DESCRIPTION
**Pull request by oscardaly**
### Description

Changes made:
- Add more detail to the GraphQL docs
- Add images with example code to show how it is used

I was doing a talk on GraphQL at work and when researching came across the docs that Codecademy had and found that they were very limited so when I finished the talk I took the notes I used and updated the Codecademy docs with these to be a bit more thorough.

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by oscardaly** [comment]
There's a test failing in the pipeline but it appears to be due to the configuration of a test file that I didn't make any changes to so I'm hesitant to try and fix it.

**Comment by yangc95** [comment]
> There's a test failing in the pipeline but it appears to be due to the configuration of a test file that I didn't make any changes to so I'm hesitant to try and fix it.

@oscardaly feel free to disregard the failed test and only push up changes for the files we're working on for this PR. We'll help with the tests prior to merge!

**Comment by oscardaly** [comment]
> > There's a test failing in the pipeline but it appears to be due to the configuration of a test file that I didn't make any changes to so I'm hesitant to try and fix it.
> 
> @oscardaly feel free to disregard the failed test and only push up changes for the files we're working on for this PR. We'll help with the tests prior to merge!

Okay, thank you. That's the pr finished and ready for review, there are no more changes or files to uplodad.

File: content/open-source/concepts/graphql/graphql.md
Comment (yangc95) [inline content/open-source/concepts/graphql/graphql.md:18]:
```suggestion
[**GraphQL**](https://graphql.org/) is an open-source query language that allows developers to consume, construct, and create [APIs](https://www.codecademy.com/resources/docs/general/api) in their own unique way. It allows us to perform actions with API's via a query language, similar to using SQL when performing CRUD (Create, Read, Update, and Delete) operations on a database. A GraphQL service is created by defining types, with different fields available on those types, and then providing functions for each of those fields on how to resolve them. GraphQL supports multiple languages including JavaScript, TypeScript, Java, C#, Go, Python, and Ruby.
```

File: content/open-source/concepts/graphql/graphql.md
Comment (yangc95) [inline content/open-source/concepts/graphql/graphql.md:26]: Can we add a sentence to preface the bullet points below?

File: content/open-source/concepts/graphql/graphql.md
Comment (yangc95) [inline content/open-source/concepts/graphql/graphql.md:32]: Can we remove any pronouns (e.g., we and you) and adjust accordingly for these points in addition to the rest of the additions in this entry?

File: content/open-source/concepts/graphql/graphql.md
Comment (yangc95) [inline content/open-source/concepts/graphql/graphql.md:50]: This is a structural comment: wrap any code in backticks like `Character` and `String`, instead of double quotes.

File: content/open-source/concepts/graphql/graphql.md
Comment (yangc95) [inline content/open-source/concepts/graphql/graphql.md:77]: Can we change this from bullets to paragraph format?

File: content/open-source/concepts/graphql/graphql.md
Comment (yangc95) [inline content/open-source/concepts/graphql/graphql.md:86]: Can we either preface these bullet points or turn into paragraph format?

**Review by yangc95 — COMMENTED** [review]
@oscardaly thank you for this enhancement! I conducted a first review of which consists mainly of structural suggestions and comments.

Please tag me once all the comments have been resolved or if you have any questions 🙂

**Comment by oscardaly** [comment]
> @oscardaly thank you for this enhancement! I conducted a first review of which consists mainly of structural suggestions and comments.
> 
> Please tag me once all the comments have been resolved or if you have any questions 🙂

@yangc95 Thank you for the feedback :) I believe that should hopefully be everything addressed

**Review by yangc95 — APPROVED** [review]
@oscardaly Nice! Thanks for looking over those comments. This is ready for a second review ☺️ 

I also made a few link inserts and grammar edits here: https://github.com/Codecademy/docs/pull/1462/commits/54ffcacd545f85c0449d89f11ede3187e92f4bd7

**Comment by oscardaly** [comment]
> @oscardaly Nice! Thanks for looking over those comments. This is ready for a second review ☺️
> 
> I also made a few link inserts and grammar edits here: [54ffcac](https://github.com/Codecademy/docs/commit/54ffcacd545f85c0449d89f11ede3187e92f4bd7)

Been over a week now since your approval, do we need to tag someone in this pr to request their review?

**Comment by yangc95** [comment]
> do we need to tag someone in this pr to request their review?

@oscardaly thanks for following up and for your patience. If there is an excessive delay, I can go ahead and request a review. In any case, a maintainer should hopefully get to your PR within the coming week to give you a final review. 🙂

File: content/open-source/concepts/graphql/graphql.md
Comment (KTom101) [inline content/open-source/concepts/graphql/graphql.md:42]:
```suggestion
In a system like REST, only a single set of arguments can be passed in: the query parameters and [URL](https://www.codecademy.com/resources/docs/general/url) segments of the request. But in GraphQL, every field and nested object has its own set of arguments, making GraphQL a complete replacement for making multiple API fetches. By passing arguments into scalar fields, data transformations can be implemented on the server rather than on every individual client, which can be more efficient.
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@oscardaly I only have a few minor changes for this.

**Comment by oscardaly** [comment]
> @oscardaly I only have a few minor changes for this.

Thanks for the suggestions! That's everything implemented

**Review by KTom101 — APPROVED** [review]
@oscardaly LGTM 👍 Added small change [here](https://github.com/Codecademy/docs/pull/1462/commits/fca5ae8e024e1a54161153863f9677e200410a0a)

**Comment by KTom101** [comment]
@oscardaly Thanks for the contribution, the changes are now [live](https://www.codecademy.com/resources/docs/open-source/graphql)! Please feel free to make more contributions 🙏 You can take a look at the [issues tab](https://github.com/Codecademy/docs/issues) or add something else that you think should be added to Codecademy Docs 😁
Screen Shot 2023-02-07 at 6 42 22 PM

**Pull request merged**
